src/go/rpk/pkg/cli/cmd/redpanda/admin/selftest/start.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package selftest
+
+import (
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+
+	"github.com/docker/go-units"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewStartCommand(fs afero.Fs) *cobra.Command {
+	var (
+		yes            bool
+		diskDurationMs int
+		netDurationMs  int
+		onNodes        []int
+		onlyDisk       bool
+		onlyNetwork    bool
+	)
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Starts a new self test run",
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			// Load config settings
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			// Sanitize input
+			if diskDurationMs <= 0 || netDurationMs <= 0 {
+				out.Die("Negative duration failed validation")
+			}
+
+			// Warn user before continuing, proceed only via explicit signal
+			if !yes {
+				confirmed, err := out.Confirm("Invoking the redpanda self tests may put excessive pressure on your system. It attempts to benchmark disk and network hardware to find their max throughputs. Its advised to not start the test if there are already large workloads running on the system.")
+				out.MaybeDie(err, "unable to confirm user confirmation: %v", err)
+				if !confirmed {
+					out.Exit("self-test start was cancelled")
+				}
+			}
+
+			// Create new HTTP client for communication w/ admin server
+			cl, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// Using cmd line args, assemble self_test_start_request body
+			tests := assembleTests(onlyDisk, onlyNetwork, uint(diskDurationMs), uint(netDurationMs))
+
+			// Make HTTP POST request to leader that starts the actual test
+			tid, err := cl.StartSelfTest(cmd.Context(), onNodes, tests)
+			out.MaybeDie(err, "unable to start self test: %v", err)
+			fmt.Printf("Redpanda self-test has started, test identifier: %v\n", tid)
```

Comment:
wdyt adding a hint on the next step to check the status:

```
fmt.Printf("Redpanda self-test has started, test identifier: %v, To check the status run:\n    rpk redpanda admin self-test status", tid)
```

Reply:
This is really nice +1